src/reference/config/bindings/binding-asyncapi.md

```diff
@@ -0,0 +1,383 @@
+---
+shortTitle: asyncapi
+description: Zilla runtime asyncapi binding
+category:
+  - Binding
+tag:
+  - Server
+  - Proxy
+  - Client
+---
+
+# asyncapi Binding
+
+Zilla runtime `asyncapi` binding.
+
+```yaml {2}
+name: zilla-mqtt-kafka-broker
+bindings:
+  asyncapi_server:
+    type: asyncapi
+    kind: server
+    options:
+      specs:
+        my-mqtt-api-spec: mqtt/asyncapi.yaml
+    exit: asyncapi_proxy
+  asyncapi_proxy:
+    type: asyncapi
+    kind: proxy
+    options:
+      specs:
+        my-mqtt-api-spec: mqtt/asyncapi.yaml
+        my-kafka-api-spec: kafka/asyncapi.yaml
+      mqtt_kafka:
```

Comment:
@jfallows I am curious about this property name having an `_` underscore instead of a `-` dash. right now, unless there is an example I haven't seen yet, we use underscores in binding names and dashes in property names.

```
bindings:
  mqtt_kafka:
    type: mqtt-kafka
```

I am just thinking through what precedence this will set for future composite bindings. Additionally, as I am typing this wondering how references to this binding will show up in logs and metrics, and the like, where it might be good to still have a mechanism to use a user provided nameing scheme or override. 

```
  asyncapi_proxy:
    type: asyncapi
    kind: proxy
    options:
      mqtt-kafka:
        name: mqtt_kafka
```

Reply:
@vordimous Yeah, agreed this option naming is inconsistent from the outer perspective, and we might want to change from `mqtt_kafka` to `mqtt-kafka` here since it matters more what the type is than how we typically name the bindings internally.

I think we need to try hard not to open up too much of the internal details of composite namespace(s), as the more we do, the harder it will be to make changes later without it becoming an incompatible breaking change.

Reply:
Agreed, less is more! I think `mqtt-kafka` is a little more clear since it matches the binding type and visually doesn't look like a "different" type of thing.